salt/utils/yamldumper.py

```diff
@@ -69,31 +58,26 @@ def represent_undefined(dumper, data):
     return dumper.represent_scalar("tag:yaml.org,2002:null", "NULL")
 
 
-OrderedDumper.add_representer(OrderedDict, represent_ordereddict)
-SafeOrderedDumper.add_representer(OrderedDict, represent_ordereddict)
-SafeOrderedDumper.add_representer(None, represent_undefined)
-
-OrderedDumper.add_representer(
-    collections.defaultdict, yaml.representer.SafeRepresenter.represent_dict
-)
-SafeOrderedDumper.add_representer(
-    collections.defaultdict, yaml.representer.SafeRepresenter.represent_dict
-)
-OrderedDumper.add_representer(
-    salt.utils.context.NamespacedDictWrapper,
-    yaml.representer.SafeRepresenter.represent_dict,
-)
-SafeOrderedDumper.add_representer(
-    salt.utils.context.NamespacedDictWrapper,
-    yaml.representer.SafeRepresenter.represent_dict,
-)
-
-OrderedDumper.add_representer(
-    "tag:yaml.org,2002:timestamp", OrderedDumper.represent_scalar
-)
-SafeOrderedDumper.add_representer(
-    "tag:yaml.org,2002:timestamp", SafeOrderedDumper.represent_scalar
-)
+# OrderedDumper does not inherit from SafeOrderedDumper, so any applicable
+# representers added to SafeOrderedDumper must also be explicitly added to
+# OrderedDumper.
+for D in (SafeOrderedDumper, OrderedDumper):
+    # This default registration matches types that don't match any other
+    # registration, overriding PyYAML's default behavior of raising an
+    # exception.  This representer instead produces null nodes.
+    #
+    # TODO: Why does this registration exist?  Isn't it better to raise an
+    # exception for unsupported types?
+    D.add_representer(None, represent_undefined)
```

Comment:
This representer is only meant for the SafeOrderedDumpter not the OrderedDumper. because it is meant to be "safe" alternative to blowing up aka throwing an exception. the OrderedDumper SHOULD blowup on unknown types but SafeOrderedDumper should just null gracefully.